src/Shared/LogMessagePacketBase.cs

```diff
@@ -139,7 +139,12 @@ internal enum LoggingEventType : int
         /// <summary>
         /// Event is a ResponseFileUsedEventArgs
         /// </summary>
-        ResponseFileUsedEvent = 20
+        ResponseFileUsedEvent = 20,
+
+        /// <summary>
+        /// Event is an AssemblyLoadBuildEventArgs
+        /// </summary>
+        AssemblyLoadEvent = 21,
```

Comment:
Have you tested if this works properly for a build that traverses multiple nodes? When introducing a new event args type, we typically have to implement (de)serialization in (Write/Read)Event(To/From)Stream. It's possible you can squeeze by because this is a message, but I'd want to make sure that's properly tested.

Reply:
I normally think of reflection as a bit slow but sometimes required if we don't know anything about a build event. I'm wondering if there's some opportunity to optimize this a bit if we have some WriteToStream base method in the base type and have it take a BinaryWriter (and maybe a bool) and have it return a bool to indicate that we do or don't support self-serialization? I'm imagining that if it says yes, it would just call the method afterwards. Alternatively, it could just be a query to WriteToStream() and return true/false but not do any work. What do you think?

Reply:
I totally agree the code would benefit from unifying (de)serialization techniques for cross-nodes communication - currently there are 3 different approaches: 
[MSBUILDLOGALLASSEMBLYLOADS ](https://github.com/dotnet/msbuild/blob/main/documentation/wiki/Nodes-Orchestration.md#messages-deserialization)

`ITranslatable` contract sounds as the best choice here - so ideally `BuildEventArgs` would implement and use that.

Though - that is totally out of scope of current change :-) Do you think any change to contract is needed in scope of this change? Do you want to crerate item for code/contracts refactoring?

I wouldn't worry about perf impact - reflection is definitely more costly then explicit contract implementation - though the cost is paid once per node, so I prefered to stick with current contracts

Reply:
It's once per node? I would've thought once per method per node. In any event, I can agree it's out of scope, and I can't think of anything else that is both clearly better and in scope. You can make the item or not—as you said, I don't think it'd be a high priority for us, so it might end up sitting for a while then getting closed as low priority. I think this otherwise looks good to me 🙂